vscode/src/paths.ts

```diff
@@ -90,3 +94,96 @@ export function fsPaths(): ExtensionFsPaths {
   }
   return _fsPaths;
 }
+
+const DEFAULT_IGNORES = [".git", ".vscode", "target", "node_modules"];
+const IGNORE_FILE_IN_PRIORITY_ORDER = [".konveyorignore", ".gitignore"];
+
+let _ignoreByFunction: undefined | ((path: string) => boolean);
+
+/**
+ * Find and use the right ignore settings to be able to ignore changes to a path.
+ */
+function isIgnoredBy(path: string): boolean {
+  if (!_ignoreByFunction) {
+    // Check for ignore files
+    for (const glob of IGNORE_FILE_IN_PRIORITY_ORDER) {
+      const ignoreFiles = globbySync(glob, { cwd: fsPaths().workspaceRepo });
+      if (ignoreFiles.length > 0) {
+        _ignoreByFunction = isIgnoredByIgnoreFilesSync(glob, {
+          cwd: fsPaths().workspaceRepo,
+        });
+        break;
+      }
+    }
+
+    // If no ignore files, use the default ignore patterns
+    if (!_ignoreByFunction) {
+      _ignoreByFunction = (path: string): boolean => {
+        const found = globbySync(path, {
+          cwd: fsPaths().workspaceRepo,
+          ignore: DEFAULT_IGNORES,
+        });
+        return found.length === 0;
+      };
+    }
+  }
+
+  return _ignoreByFunction(path);
+}
+
+/**
+ * Check a Uri to see if it should be ignored by a partial analysis on save.
+ */
+export const isUriIgnored = (uri: vscode.Uri): boolean => {
+  if (uri.scheme !== "file") {
+    return true;
+  }
+
+  const f = relative(fsPaths().workspaceRepo, uri.fsPath);
```

Comment:
This looks like the method we should be using when getting relativeDir for incident uris.

Reply:
Could be for sure.  It may need some extra work to go through the posix handling code to normalize input of path separators, and then to output with the platform correct path separators.  :thinking: